fix(desktop): keep Vite renderer build on Node under the Bun dev host

## Problem

On current `develop` (`1d110415417e9b4e59e52f3d50947617313c627d`), the native macOS development entrypoint fails before Electrobun launches:

```text
bun run dev:desktop -- --no-api --force-renderer
error: Cannot find module './cjs/index.cjs' from ''
Command failed: .../bun --conditions=eliza-source --import tsx .../vite.js --configLoader native build
```

`packages/app-core/scripts/lib/dev-ui-vite.mjs` documents and returns a Node-backed Vite command, but its default is `process.execPath`. Because `dev:desktop` itself runs under Bun, that default silently resolves to Bun. The identical Vite command succeeds when invoked with pinned Node.

## User impact

A clean current-develop checkout cannot reliably launch the native macOS app for local AgentRuntime development, including the supported no-Cloud-auth external runtime path.

## Acceptance criteria

- The canonical Vite command resolves a validated Node 24+ executable when its caller runs under Bun.
- Explicit invalid Node overrides still fail closed.
- Direct app, shared app, and desktop renderer entrypoints retain the same source-aware Vite arguments.
- The focused regression fails on the current implementation and passes on the repair.
- `bun run dev:desktop -- --no-api --force-renderer` builds and launches the current native macOS app against a real local AgentRuntime.
- Native shortcut invocation reaches the renderer, and one real message/reply is persisted through the local runtime.
- Required package tests, typecheck, Biome, diff, root verification, and native evidence are attached to the PR.

## Scope

Initial implementation scope is limited to:

- `packages/app-core/scripts/lib/dev-ui-vite.mjs`
- `packages/app-core/scripts/lib/dev-ui-vite.test.mjs`

No Cloud, auth, provider credential, backend product, iOS, homepage, staging, or production behavior changes.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex Desktop
Contribution skill revision: elizaOS/eliza@1d110415417e9b4e59e52f3d50947617313c627d:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [codex / apple-native-surface-lane]
<!-- eliza-computer-attribution:v1 {"provider":"openai","model":"gpt-5.6-sol","client":"Codex Desktop","skill_revision":"elizaOS/eliza@1d110415417e9b4e59e52f3d50947617313c627d:packages/skills/skills/contribute-to-eliza","status":"self-reported"} -->

--- a/packages/app-core/scripts/lib/dev-ui-vite.mjs
+++ b/packages/app-core/scripts/lib/dev-ui-vite.mjs
@@ -8,15 +8,24 @@
 import { existsSync } from "node:fs";
 import path from "node:path";
 import process from "node:process";
+import { resolveNodeExecPath } from "../run-node-runtime.mjs";
 
 export function resolveViteCommand({
   appDir,
   force = false,
-  nodePath = process.execPath,
+  nodePath,
   port,
   viteArgs = [],
 }) {
-  if (!nodePath) {
+  const resolvedNodePath =
+    nodePath === undefined
+      ? resolveNodeExecPath({
+          currentExecPath: process.execPath,
+          explicitNodePath: process.env.ELIZA_NODE_PATH,
+          platform: process.platform,
+        })
+      : nodePath;
+  if (!resolvedNodePath) {
     throw new Error("Node.js is required to run the Vite dev server.");
   }
   const viteCli = path.join(appDir, "node_modules", "vite", "bin", "vite.js");
@@ -40,5 +49,5 @@ export function resolveViteCommand({
   if (force) args.push("--force");
   if (port !== undefined) args.push("--port", String(port));
   args.push(...viteArgs);
-  return { command: nodePath, args };
+  return { command: resolvedNodePath, args };
 }

--- a/packages/app-core/scripts/lib/dev-ui-vite.test.mjs
+++ b/packages/app-core/scripts/lib/dev-ui-vite.test.mjs
@@ -1,7 +1,8 @@
 /** Verifies the development Vite subprocess uses the direct TypeScript config loader. */
 
+import { spawnSync } from "node:child_process";
 import path from "node:path";
-import { fileURLToPath } from "node:url";
+import { fileURLToPath, pathToFileURL } from "node:url";
 import { expect, test } from "vitest";
 import { resolveViteCommand } from "./dev-ui-vite.mjs";
 
@@ -28,4 +29,32 @@ test("resolveViteCommand uses workspace source while skipping config bundling", 
     "--port",
     "2138",
   ]);
+});
+
+test("resolveViteCommand stays Node-backed when its caller runs under Bun", () => {
+  const helperUrl = pathToFileURL(
+    path.join(path.dirname(fileURLToPath(import.meta.url)), "dev-ui-vite.mjs"),
+  ).href;
+  const script = `
+    import { resolveViteCommand } from ${JSON.stringify(helperUrl)};
+    const resolved = resolveViteCommand({ appDir: ${JSON.stringify(appDir)} });
+    process.stdout.write(resolved.command);
+  `;
+  const resolution = spawnSync("bun", ["--eval", script], {
+    encoding: "utf8",
+    env: { ...process.env, ELIZA_NODE_PATH: "" },
+  });
+
+  expect(resolution.status, resolution.stderr).toBe(0);
+  const nodePath = resolution.stdout.trim();
+  const runtime = spawnSync(
+    nodePath,
+    [
+      "--eval",
+      "process.stdout.write(process.versions.bun ? 'bun' : 'node:' + process.versions.node)",
+    ],
+    { encoding: "utf8" },
+  );
+  expect(runtime.status, runtime.stderr).toBe(0);
+  expect(runtime.stdout).toMatch(/^node:24\./);
 });

